
<!DOCTYPE html>
<html class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="PhyloNetworks is a Julia package for the manipulation, visualization and inference of phylogenetic networks.">
      
      
      
        <meta name="author" content="Claudia Sol&iacute;s-Lemus">
      
      
        <link rel="shortcut icon" href="../../assets/images/favicon.png">
      
      <meta name="generator" content="mkdocs-0.16.2, mkdocs-material-1.4.1">
    
    
      
        <title>Input Data for SNaQ - PhyloNetworks.jl</title>
      
    
    
      <script src="../../assets/javascripts/modernizr-56ade86843.js"></script>
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/application-76741b64bc.css">
      
        <link rel="stylesheet" href="../../assets/stylesheets/application-66fa0d9bba.palette.css">
      
    
    
      
        
        
        
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700|Roboto+Mono">
        <style>body,input{font-family:"Roboto","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono","Courier New",Courier,monospace}</style>
      
      <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons">
    
    
      <link rel="stylesheet" href="../../assets/Documenter.css">
    
    
  </head>
  
  
  
  
    <body data-md-color-primary="teal" data-md-color-accent="teal">
  
    <svg class="md-svg">
      <defs>
        
        
          <svg xmlns="http://www.w3.org/2000/svg" width="416" height="448" viewBox="0 0 416 448" id="github"><path fill="currentColor" d="M160 304q0 10-3.125 20.5t-10.75 19T128 352t-18.125-8.5-10.75-19T96 304t3.125-20.5 10.75-19T128 256t18.125 8.5 10.75 19T160 304zm160 0q0 10-3.125 20.5t-10.75 19T288 352t-18.125-8.5-10.75-19T256 304t3.125-20.5 10.75-19T288 256t18.125 8.5 10.75 19T320 304zm40 0q0-30-17.25-51T296 232q-10.25 0-48.75 5.25Q229.5 240 208 240t-39.25-2.75Q130.75 232 120 232q-29.5 0-46.75 21T56 304q0 22 8 38.375t20.25 25.75 30.5 15 35 7.375 37.25 1.75h42q20.5 0 37.25-1.75t35-7.375 30.5-15 20.25-25.75T360 304zm56-44q0 51.75-15.25 82.75-9.5 19.25-26.375 33.25t-35.25 21.5-42.5 11.875-42.875 5.5T212 416q-19.5 0-35.5-.75t-36.875-3.125-38.125-7.5-34.25-12.875T37 371.5t-21.5-28.75Q0 312 0 260q0-59.25 34-99-6.75-20.5-6.75-42.5 0-29 12.75-54.5 27 0 47.5 9.875t47.25 30.875Q171.5 96 212 96q37 0 70 8 26.25-20.5 46.75-30.25T376 64q12.75 25.5 12.75 54.5 0 21.75-6.75 42 34 40 34 99.5z"/></svg>
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="drawer">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="search">
    <label class="md-overlay" data-md-component="overlay" for="drawer"></label>
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        
          <a href="../.." title="PhyloNetworks.jl" class="md-logo md-header-nav__button">
            <img src="../../snaq_small.png" width="24" height="24">
          </a>
        
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <span class="md-flex__ellipsis md-header-nav__title">
          
            
              
                <span class="md-header-nav__parent">
                  Manual
                </span>
              
            
            Input Data for SNaQ
          
        </span>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          <label class="md-icon md-icon--search md-header-nav__button" for="search"></label>
          
<div class="md-search" data-md-component="search">
  <div class="md-search__overlay"></div>
  <div class="md-search__inner">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" placeholder="Search" accesskey="s" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query">
      <label class="md-icon md-search__icon" for="search"></label>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result">
          <div class="md-search-result__meta" data-md-lang-result-none="No matching documents" data-md-lang-result-one="1 matching document" data-md-lang-result-other="# matching documents">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
        
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <div class="md-header-nav__source">
          
            


  


  <a href="https://github.com/crsl4/PhyloNetworks.jl" title="Go to repository" class="md-source" data-md-source="github">
    
      <div class="md-source__icon">
        <svg viewBox="0 0 24 24" width="24" height="24">
          <use xlink:href="#github" width="24" height="24"></use>
        </svg>
      </div>
    
    <div class="md-source__repository">
      GitHub
    </div>
  </a>

          
        </div>
      </div>
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
      
      <main class="md-main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="drawer">
    
      <i class="md-logo md-nav__button">
        <img src="../../snaq_small.png">
      </i>
    
    PhyloNetworks.jl
  </label>
  
    <div class="md-nav__source">
      


  


  <a href="https://github.com/crsl4/PhyloNetworks.jl" title="Go to repository" class="md-source" data-md-source="github">
    
      <div class="md-source__icon">
        <svg viewBox="0 0 24 24" width="24" height="24">
          <use xlink:href="#github" width="24" height="24"></use>
        </svg>
      </div>
    
    <div class="md-source__repository">
      GitHub
    </div>
  </a>

    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href="../.." title="Home" class="md-nav__link">
      Home
    </a>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-2" type="checkbox" id="nav-2" checked>
    
    <label class="md-nav__link" for="nav-2">
      Manual
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-2">
        Manual
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../installation/" title="Installation" class="md-nav__link">
      Installation
    </a>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="toc">
    
      
    
    
      <label class="md-nav__link md-nav__link--active" for="toc">
        Input Data for SNaQ
      </label>
    
    <a href="./" title="Input Data for SNaQ" class="md-nav__link md-nav__link--active">
      Input Data for SNaQ
    </a>
    
      
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#tutorial-data-gene-trees" title="Tutorial data: gene trees" class="md-nav__link">
    Tutorial data: gene trees
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#tutorial-data-quartet-cfs" title="Tutorial data: quartet CFs" class="md-nav__link">
    Tutorial data: quartet CFs
  </a>
  
</li>
      
      
      
    </ul>
  
</nav>
    
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../ticr_howtogetQuartetCFs/" title="TICR pipeline" class="md-nav__link">
      TICR pipeline
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../snaq_plot/" title="Network estimation and display" class="md-nav__link">
      Network estimation and display
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../fixednetworkoptim/" title="Fixed Network" class="md-nav__link">
      Fixed Network
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../expectedCFs/" title="Extract Expected CFs" class="md-nav__link">
      Extract Expected CFs
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../bootstrap/" title="Bootstrap" class="md-nav__link">
      Bootstrap
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../multiplealleles/" title="Multiple Alleles" class="md-nav__link">
      Multiple Alleles
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../trait_tree/" title="Trait Evolution" class="md-nav__link">
      Trait Evolution
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-3" type="checkbox" id="nav-3">
    
    <label class="md-nav__link" for="nav-3">
      Library
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-3">
        Library
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../lib/public/" title="Public" class="md-nav__link">
      Public
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../lib/internals/" title="Internals" class="md-nav__link">
      Internals
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#tutorial-data-gene-trees" title="Tutorial data: gene trees" class="md-nav__link">
    Tutorial data: gene trees
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#tutorial-data-quartet-cfs" title="Tutorial data: quartet CFs" class="md-nav__link">
    Tutorial data: quartet CFs
  </a>
  
</li>
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                  <a href="https://github.com/crsl4/PhyloNetworks.jl/edit/master/docs/man/inputdata.md" title="Edit this page" class="md-icon md-content__icon">edit</a>
                
                
                <p><a id='Input-for-SNaQ-1'></a></p>
<h1 id="input-for-snaq">Input for SNaQ</h1>
<p>SNaQ is a method implemented in the package to estimate a phylogenetic network from multiple molecular sequence alignments. There are two alternatives for the input data:</p>
<ol>
<li>A list of estimated gene trees for each locus, which can be obtained using</li>
</ol>
<p><a href="http://mrbayes.sourceforge.net">MrBayes</a> or <a href="http://sco.h-its.org/exelixis/software.html">RAxML</a>. Or:</p>
<ol>
<li>A table of concordance factors (CF) for each 4-taxon subset which can be</li>
</ol>
<p>obtained from <a href="http://www.stat.wisc.edu/~ane/bucky/">BUCKy</a>,   to account for gene tree uncertainty</p>
<p>This <a href="https://github.com/nstenz/TICR">pipeline</a> can be used to obtain the table of  quartet CF needed as input for SNaQ. The pipeline starts from the sequence alignments, runs MrBayes and then BUCKy (both parallelized), producing the table of estimated CFs and their credibility intervals. Additional details on this <a href="../ticr_howtogetQuartetCFs/#TICR-pipeline-1">TICR pipeline</a> describe how to insert data at various stages (e.g. after running MrBayes on each locus).</p>
<p><a id='Tutorial-data:-gene-trees-1'></a></p>
<h2 id="tutorial-data-gene-trees">Tutorial data: gene trees</h2>
<!–The examples files for this section can be found within the PhyloNetworks folder, typically in your *HOME/.julia/v0.4/PhyloNetworks/examples/*. However, links to the files are also included below. –> We suggest that you create a special directory for running these examples, where input files can be downloaded and where output files will be created (with estimated networks for instance). Enter this directory and run Julia from there.

<p>Suppose you have a file with a list of gene trees in parenthetical format called <em>treefile.txt</em>. You can access the example file of input trees <a href="https://github.com/crsl4/PhyloNetworks/blob/master/examples/treefile.txt">here</a> or <a href="https://raw.githubusercontent.com/crsl4/PhyloNetworks/master/examples/treefile.txt">here</a> for easier download.</p>
<p>Do not copy-paste into a "smart" text-editor. Instead, save the file directly into your working directory using "save link as" or "download linked file as". This file contains 10 trees, each in parenthetical format on 6 taxa like this:</p>
<p>(6:2.728,((3:0.655,5:0.655):1.202,(1:0.881,(2:0.783,4:0.783):0.098):0.976):0.871);</p>
<p>If 'treefile.txt' is in your working directory, you can view its content within Julia:</p>
<pre><code class="julia">less(&quot;treefile.txt&quot;)
</code></pre>

<p>Just type <code>q</code> to quit viewing this file. You could read in these 10 trees and visualize the third one (say) like this:</p>
<pre><code class="julia">tentrees = readMultiTopology(&quot;treefile.txt&quot;)
tentrees[3]
plot(tentrees[3])
</code></pre>

<p>To read in all gene trees and directly summarize them by a list of quartet CFs (proportion of input trees with a given quartet):</p>
<pre><code class="julia">d=readTrees2CF(&quot;treefile.txt&quot;, CFfile=&quot;tableCFall.txt&quot;)
less(&quot;tableCFall.txt&quot;)
</code></pre>

<p><code>less("tableCFall.txt")</code> lets you see the content of the newly created file "tableCFall.txt", within Julia. Again, type <code>q</code> to quit viewing this file.</p>
<p>If instead of all the 4-taxon subsets, you just want to use a random sample of 10 4-taxon subsets:</p>
<pre><code class="julia">d=readTrees2CF(&quot;treefile.txt&quot;, whichQ=&quot;rand&quot;, numQ=10, CFfile=&quot;tableCF10.txt&quot;)
</code></pre>

<p>Be careful to use a numQ value smaller than the total number of possible 4-taxon subsets, which is <em>n choose 4</em> on <em>n</em> taxa (e.g. 15 on 6 taxa). To get a predictable random sample, you may set the seed with <code>srand(12321)</code> (for instance) prior to sampling the quartets as above.</p>
<p><a id='Tutorial-data:-quartet-CFs-1'></a></p>
<h2 id="tutorial-data-quartet-cfs">Tutorial data: quartet CFs</h2>
<p>If you already have a table of quartet concordance factor (CF) values in a file <em>tableCF.txt</em> in this format</p>
<table>
<thead>
<tr>
<th align="left">Taxon1</th>
<th align="left">Taxon2</th>
<th align="left">Taxon3</th>
<th align="left">Taxon4</th>
<th align="left">CF12_34</th>
<th align="left">CF13_24</th>
<th align="left">CF14_23</th>
</tr>
</thead>
<tbody>
<tr>
<td align="left">A</td>
<td align="left">B</td>
<td align="left">E</td>
<td align="left">D</td>
<td align="left">0.4</td>
<td align="left">0.6</td>
<td align="left">0.0</td>
</tr>
</tbody>
</table>
<p>you would read it like this:</p>
<pre><code class="julia">d=readTableCF(&quot;tableCF.txt&quot;) # one step only: read file and convert to 'DataCF' object
# or in 2 steps:
using DataFrames
dat = readtable(&quot;tableCF.txt&quot;) # read file into a 'DataFrame' object
d = readTableCF(dat)           # convert DataFrame into DataCF
</code></pre>

<p>You can access this example file <a href="https://github.com/crsl4/PhyloNetworks/blob/master/examples/tableCF.txt">here</a> or <a href="https://raw.githubusercontent.com/crsl4/PhyloNetworks/master/examples/tableCF.txt">here</a>.</p>
<p>Columns need to be in the right order. If you have the information on the number of genes used for each 4-taxon subset, you can add this information in a column named "ngenes".</p>
<p>If you have a tree <em>startTree.txt</em> in parenthetical format to use as starting point for the optimization, you can read it with</p>
<pre><code class="julia">T=readTopology(&quot;startTree.txt&quot;)
</code></pre>

<p>You can access this example file <a href="https://github.com/crsl4/PhyloNetworks/blob/master/examples/startTree.txt">here</a> (raw file <a href="https://raw.githubusercontent.com/crsl4/PhyloNetworks/master/examples/startTree.txt">here</a>).</p>
<p>If the topology <code>T</code> is to be used as a starting tree for SNaQ, it needs to be of level 1. To make sure that it is, you may read it in with <code>readTopologyLevel1</code> (which may also unroot the tree, resolve polytomies, replace missing branch lengths by 1 for starting values etc.):</p>
<pre><code class="julia">T=readTopologyLevel1(&quot;startTree.txt&quot;)
</code></pre>

<p>Note that all trees and all networks with 1 hybridization are of level 1.</p>
<p>Next: <a href="../snaq_plot/#Getting-a-Network-1">Getting a Network</a></p>
                
                  
                
              
              
                
              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid">
        
          <a href="../installation/" title="Installation" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i>
            </div>
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                Installation
              </span>
            </div>
          </a>
        
        
          <a href="../ticr_howtogetQuartetCFs/" title="TICR pipeline" class="md-flex md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                TICR pipeline
              </span>
            </div>
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-forward md-footer-nav__button"></i>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
        powered by
        <a href="http://www.mkdocs.org" title="MkDocs">MkDocs</a>
        and
        <a href="http://squidfunk.github.io/mkdocs-material/" title="Material for MkDocs">
          Material for MkDocs</a>
      </div>
      
        
      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../../assets/javascripts/application-3964c0cb56.js"></script>
      <script>app.initialize({url:{base:"../.."}})</script>
      
        <script src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
      
        <script src="../../assets/mathjaxhelper.js"></script>
      
    
    
      
    
  </body>
</html>